hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -796,6 +796,15 @@ def update_shrink_target(self, new_target):
         assert isinstance(new_target, ConjectureResult)
         if self.shrink_target is not None:
             self.shrinks += 1
+            # If we are just taking a long time to shrink we don't want to
+            # trigger this heuristic, so whenever we shrink successfully
+            # we give ourselves a bit of breathing room to make sure we
+            # would find a shrink that took that long to find the next
+            # time.
+            self.max_stall = max(
+                self.max_stall, (self.calls - self.calls_at_last_shrink) * 2
+            )
```

Comment:
It took me a moment to realise that we don't need an upper bound, because either we stop making progress and `max_stall` is appropriate, or we are making very slow progress and the five-minute timeout will handle that.

Noting this in the comment would probably be helpful for future readers.